SPLAB-UFCG/Moenda
# Add moenda engine Node API

## Motivation

I am developing a tool similar to linter and would like to integrate moenda into the system to run the rules of the tool, but currently moenda only works for a CLI client, this PR creates new modules enabling the use of Moenda as an API.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/SPLAB-UFCG/Moenda/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

1. Go to `tests` folder
`cd tests/`

2. Run `node test.js` 

3. You should get an output similar to that
```js
[
// ....
  {
    ruleName: 'require-structure',
    filePath: '/Users/fanny.vieira/dev/ufcg/Moenda/tests/files/teste.md',
    column: 0,
    line: 216,
    detail: 'Your section is not following the recommended structure',
    context: '`React.Children` {#reactchildren}',
    severity: 2
  },
  {
    ruleName: 'require-structure',
    filePath: '/Users/fanny.vieira/dev/ufcg/Moenda/tests/files/teste.md',
    column: 0,
    line: 291,
    detail: 'Your section is not following the recommended structure',
    context: '`React.createRef` {#reactcreateref}',
    severity: 2
  },
  {
    ruleName: 'require-structure',
    filePath: '/Users/fanny.vieira/dev/ufcg/Moenda/tests/files/teste.md',
    column: 0,
    line: 296,
    detail: 'Your section is not following the recommended structure',
    context: '`React.forwardRef` {#reactforwardref}',
    severity: 2
  }
]
finish
```


## Related PRs or/and Issues

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/SPLAB-UFCG/Moenda, and link to your PR here.)
